### DISCUSSION TITLE
Backup: Only new photos

### DISCUSSION THREAD

**Discussion by reitbauer** [immich-app/immich#7657] [Feature Request]
### The feature

It would be good if it were possible to specify that only photos and videos from today should be uploaded when backing up photos.
This would be very important when changing smartphones, for example.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by patrontheo** [comment]
I agree. 
I just switched to immich and added all my existing photos as an external library.
If I activate mobile sync, all the photos I have on my phone will be uploaded to immich standard library, right ? This would lead to a lot of duplicates.. 
The only workaround I found is to only backup a specific album, and put all my new photos in this album, but that's not really handy..
Is there a better solution @alextran1502 ?
(Thank you for the amazing work !!)

**Reply by alextran1502** [reply]
We have plan to work on this

**Reply by patrontheo** [reply]
Thanks :)
In the meantime, I found this shortcut: https://thesweetsetup.com/how-to-use-shortcuts-to-add-photos-to-a-specific-album/
I'll try to make it run automatically once a day, and it should add new photos to the given album.

**Reply by patrontheo** [reply]
Here is a shortcut I created to put photos taken after a given date to a backup album: https://www.icloud.com/shortcuts/53cdef71c0144e1296fbfc3467dc2bd3
You can change the date, and add an automation to run this shortcut one / multiple time(s) a day :)

**Reply by ElXando** [reply]
> We have plan to work on this

Any ETA? Just installed Immich and can't really start photo uploading without this!

**Reply by alextran1502** [reply]
@ElXando the upload/sync album mechanism is in the latest release unless I am mistake your request is for something else?

**Reply by patrontheo** [reply]
@alextran1502 I think he wants to sync the main library but only images more recent than a given date.

**Reply by alextran1502** [reply]
Ah, then we have yet to make a plan for that. The comment above was explicitly for uploading to an album, which has been implemented in the most recent release

**Reply by ElXando** [reply]
@alextran1502 Yes sorry I meant the ability to either only upload photos from a specific date OR to batch mark all photos in an album as having been already uploaded. My camera has 20k+ photos in it which are already on my server via Amazon Photos...

**Comment by cristian98149** [comment]
Could be very useful to compare local photo hash with server photos hashes in order to detect which image needs to be backed-up insted of using device ID as I understand from this answer (https://github.com/immich-app/immich/issues/9495#issuecomment-2113818459).

Thank you all.

**Comment by CarlvO** [comment]
Plus one on this. Use case: I migrated all my pics from a self-baked solution using Foldersync and an SMB server. All pics I made before today are therefore already on my Immich box. Therefore autosync should be able to start for all new pics I make. If the app has its own database of sync-ed pics (which I suppose it does) it could have a toggle to sync new pics. If activated, it marks all pics in the indicated folders as synced and the rest should remain the same.

Thanks for a great photo organizer!

**Comment by bo0tzz** [comment]
#4169
